questionnaire/dataset/dataset-service.js

```diff
@@ -21,42 +72,84 @@ function createDatasetService({
+                    const existingAnswer = dataset.get(questionId);
+
+                    if (existingAnswer !== undefined) {
+                        const mergedAnswers = mergeArrays(existingAnswer, answer);
 
-                    if (dataset.has(questionId)) {
-                        const existingAnswer = dataset.get(questionId);
-
-                        // Answers for a specific question id can be collected over multiple sections.
-                        // If previous answers have already be processed for an id, use an appropriate
-                        // merge strategy for the given data type e.g. push to an array, concatenate strings with a delimiter
-                        // ONLY SUPPORTS ARRAYS AT THE MOMENT
-                        if (Array.isArray(existingAnswer)) {
-                            // Ensure answers are of the same type
-                            if (Array.isArray(answer)) {
-                                existingAnswer.push(...answer);
-                            } else {
-                                throw new VError(
-                                    `Question id "${questionId}" found more than once with different answer types. Unable to combine type "array" with "${typeof answer}"`
-                                );
-                            }
-                        } else {
-                            throw new VError(
-                                `Question id "${questionId}" found more than once with unsupported type "${typeof existingAnswer}". Only arrays can be used to combine answers for a single id`
-                            );
-                        }
+                        dataset.set(questionId, mergedAnswers);
                     } else {
                         dataset.set(questionId, answer);
                     }
                 });
             }
         });
 
-        return [
-            {
-                type: 'dataset',
-                id: 0,
-                attributes: Object.fromEntries(dataset)
+        return Object.fromEntries(dataset);
+    }
+
+    function getHierachicalDataView(questionnaire) {
+        const {progress, sections, answers} = questionnaire;
+        const dataset = new Map();
+
+        progress.forEach(sectionId => {
+            const questionAnswers = answers[sectionId];
+
+            if (questionAnswers !== undefined) {
+                const sectionDefinition = sections[sectionId];
+                const section = createSection({sectionDefinition});
+                const attributes = section.getAttributesByData(questionAnswers);
+
+                attributes.forEach(attribute => {
+                    const existingAttribute = dataset.get(attribute.id);
+
+                    if (existingAttribute !== undefined) {
+                        mergeAttributeValues(existingAttribute, attribute);
+                    } else {
+                        const mutatedAttribute = applyGlobalAttributeDetails(
```

Comment:
are the code blocks within this if/else always mutually exclusive? I must not be understanding this logic as it appears to be the `if` deals with the munging of the phy-inj stuff... and the `else` deals with the retrieval of the global meta title (also related to phy-inj.

Reply:
@armoj If there is an existing attribute it'll merge them, else it'll apply any global attribute data that has been defined.

So for physical injuries, it'll get the global label initially. Then when it encounters it again it'll merge the values.

Reply:
Ok makes sense. So this code programmatically assumes that if there is a "global meta title", then there will be more than 1 attribute to merge together here (e.g. exactly like the physical injuries